### PR TITLE
Add types

### LINKED ISSUE #8: Typings
### Subject of the feature

Request: Typescript .d.ts typings

### Problem

Not urgent. But typings would be useful. I have since mocked them to get by.

### Alternatives

I have checked the typings on DefinitelyTyped for [unist-util-is](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/95a4800caa5b80801f02ddf400e03698b9747042/types/unist-util-is/index.d.ts) and tried to adapt them to this module. But creating typings is not my strong point.

Thank you for your work.

### PR DESCRIPTION
**Pull request by shanehandley**
Adds typings according to the format from `unist-builder`

See https://github.com/syntax-tree/unist-util-inspect/issues/8#issuecomment-546776250

Closes #8

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,10 +18,12 @@
   "files": [
     "index.js"
   ],
+  "types": "./index.d.ts",
```
Comment (ChristianMurphy) [inline package.json]: the typing is in the `types` folder

```suggestion
  "types": "types/index.d.ts",
```

File: package.json [hunk 2]
```diff
@@ -18,10 +18,12 @@
   "files": [
     "index.js"
```
Comment (ChristianMurphy) [inline package.json]: The typings would need to be included in the files release to go out with a release.

```suggestion
    "index.js",
    "types/index.d.ts"
```

File: types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// Type definitions for unist-util-inspect 4.1
+// Project: https://github.com/syntax-tree/unist-util-inspect/#readme
+// Definitions by: Shane Handley <https://github.com/shanehandley>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
```
Comment (ChristianMurphy) [inline types/index.d.ts]: Now that the typings are included in the package, these fields are no longer needed

```ts
// Type definitions for unist-util-inspect 4.1
// Project: https://github.com/syntax-tree/unist-util-inspect/#readme
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
```

---

This should be moved to _package.json_ in the `contributors` field.

```ts
// Definitions by: Shane Handley <https://github.com/shanehandley>
```

File: types/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "compilerOptions": {
+      "lib": ["es2015"],
+      "strict": true,
+      "baseUrl": ".",
+      "paths": {
+          "unist-util-inspect": ["index.d.ts"]
+      }
+  },
```
Comment (ChristianMurphy) [inline types/tsconfig.json]: This comma is causing an invalid json error

```suggestion
  }
```

File: types/tslint.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "dtslint/dt.json"
```
Comment (ChristianMurphy) [inline types/tslint.json]: this causes the error `Error: If 'tslint.json' is present, it should extend "dtslint/dtslint.json"`

```suggestion
  "extends": "dtslint/dtslint.json"
```

File: package.json [hunk 6]
```diff
@@ -18,10 +18,12 @@
...
   ],
+  "types": "./index.d.ts",
   "dependencies": {
     "is-empty": "^1.0.0"
   },
   "devDependencies": {
```
Comment (ChristianMurphy) [inline package.json]: Could `dtslint` be added as a dev dependency, and `dtslint types` be added as a `test-types` script?

**Comment by shanehandley** [comment]
Thanks @ChristianMurphy, sorry I rushed this initially. I've made the requested changes.

**Pull request merged**